### PR TITLE
docs(security): add comprehensive security documentation and test suite

### PR DESCRIPTION
**Pull request by jpoley**
## Summary

Implements task-218 and task-219: Security documentation and test suite for `/jpspec:security` workflow.

### Documentation (task-218)

- **Security Quickstart Guide** (`docs/guides/security-quickstart.md`)
  - Quick start with scan/triage/fix/audit workflow
  - Configuration examples
  - Common workflows and troubleshooting

- **Command Reference** (`docs/reference/jpspec-security-commands.md`)
  - Full CLI command documentation
  - All options with examples
  - Slash command descriptions

- **CI/CD Integration Examples** (`docs/guides/security-cicd-integration.md`)
  - GitHub Actions (basic and full pipeline)
  - GitLab CI with security dashboard
  - Jenkins (Jenkinsfile examples)
  - Azure DevOps pipelines
  - CircleCI configuration
  - Pre-commit hook integration

- **Threat Model** (`docs/reference/security-threat-model.md`)
  - Trust boundaries
  - Threat analysis with mitigations
  - Known limitations

- **Privacy Policy** (`docs/reference/security-privacy-policy.md`)
  - AI data handling
  - Scanner data policies
  - Compliance considerations

- **Custom Rules Guide** (`docs/guides/security-custom-rules.md`)
  - Semgrep rule writing
  - Pattern examples
  - Testing and integration

### Test Suite (task-219)

85 unit tests covering:

- **test_models.py**: Finding, Location, Severity, Confidence
- **test_orchestrator.py**: Scanner orchestration, deduplication
- **test_semgrep.py**: Semgrep adapter with mocked output

Test coverage includes:
- Fingerprinting and merging
- Deduplication logic
- Error handling
- SARIF conversion
- Configuration passing

## Test Plan

- [x] All 85 tests pass
- [x] Ruff lint passes
- [x] Ruff format passes
- [x] DCO sign-off present

## Acceptance Criteria

task-218:
- [x] Security Quickstart Guide
- [x] Command Reference
- [x] CI/CD Integration Examples
- [x] Threat Model and Limitations
- [x] Privacy Policy for AI
- [x] Custom Rule Writing Guide

task-219:
- [x] Unit tests for scanner with mocked output
- [x] Integration tests for orchestrator
- [x] Model tests with edge cases
- [x] Full coverage of core security module

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docs/reference/jpspec-security-commands.md [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+| `SPECIFY_SECURITY_SCANNER` | Default scanner | `semgrep` |
+| `SPECIFY_SECURITY_OUTPUT` | Default output directory | `./security-reports` |
+| `SEMGREP_APP_TOKEN` | Semgrep App token | None |
+| `GITHUB_TOKEN` | GitHub token for SARIF upload | None |
+
+---
+
+## Configuration File
+
+See [Security Configuration Reference](./security-configuration.md) for complete `.specify/security.yml` schema.
```
[the remark this answers is not in the document]
Comment (jpoley) [reply docs/reference/jpspec-security-commands.md]: fix me
